guillotina_amqp/worker.py

```diff
@@ -132,13 +131,15 @@ def num_running(self):
         # Cancelation
         if await ts.is_canceled():
             logger.warning(f"Task {_id} has already been canceled")
-            raise TaskAlreadyCanceled(_id)
+            # Ack so that canceled job is removed from main queue
+            await channel.basic_client_ack(delivery_tag=envelope.delivery_tag)
+            return
 
         try:
             await ts.acquire()
         except TaskAlreadyAcquired:
             logger.warning(f"Task {_id} is already running in another worker")
-            # TODO: ACK task
+            return
```

Comment:
We probably should ack here as well.

We always need to ack or nack a message.

If you just drop out and don't do anything with the message, the message stays unacked in the queue until that worker's amqp connection is killed. Then it goes back into the queue again.